src/analysis/retail/monk/shared/hero/MasterOfHarmony/talents/AspectOfHarmonyBase.tsx

```diff
@@ -0,0 +1,129 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  ApplyBuffEvent,
+  ApplyDebuffEvent,
+  DamageEvent,
+  EventType,
+  HealEvent,
+} from 'parser/core/Events';
+import Combatants from 'parser/shared/modules/Combatants';
+
+export interface CastInfo {
+  totalDamage: number;
+  overkill: number;
+  totalHealing: number;
+  overhealing: number;
+  numBuffs: number;
+  numDebuffs: number;
+  startTime: number;
+}
+
+class AspectOfHarmonyBaseAnalyzer extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  castEntries: CastInfo[] = [];
+  protected combatants!: Combatants;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.ASPECT_OF_HARMONY_TALENT);
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_BUFF),
+      this.onBuffApply,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_HOT),
+      this.onPeriodicApply,
+    );
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_DOT),
+      this.onPeriodicApply,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDmg);
+  }
+
+  initEntry(
```

Comment:
small nit but the name initEntry was a bit confusing at first since it is acting as both a 'get latest' and 'set'